packages/cells/src/cells/date-picker-cell.tsx

```diff
@@ -1,15 +1,87 @@
+import {
+    CustomCell,
+    CustomRenderer,
+    drawTextCell,
+    GridCellKind,
+    ProvideEditorCallback,
+} from "@glideapps/glide-data-grid";
 
 interface DatePickerCellProps {
     readonly kind: "date-picker-cell";
     readonly date: Date | undefined;
     readonly displayDate: string;
-    readonly format: "date" | "datetime-local";
+    readonly format: "date" | "time" | "datetime-local";
 }
 
+export const formatValueForHTMLInput = (format: "date" | "time" | "datetime-local", date: Date): string => {
+    if (format === "date") {
+        return date.toISOString().split("T")[0];
+    }
+    if (format === "time") {
+        return date.toISOString().split("T")[1].replace("Z", "");
+    }
+    if (format === "datetime-local") {
+        return date.toISOString().replace("Z", "");
+    }
+    return "";
+};
+
 export type DatePickerCell = CustomCell<DatePickerCellProps>;
 
+const Editor: ReturnType<ProvideEditorCallback<DatePickerCell>> = cell => {
+    const cellData = cell.value.data;
+    const { date, displayDate, format } = cellData;
+    const cellDataDate =
+        cellData.date ??
+        // now as date with adjusted timezone to allow for displayDate to be correct
+        new Date(new Date().getTime() - new Date().getTimezoneOffset() * 60000);
+    const value = formatValueForHTMLInput(format, cellDataDate);
+
+    return (
+        <input
+            required
+            style={{ minHeight: 26, border: "none", outline: "none" }}
+            type={format}
+            value={value}
+            autoFocus={true}
+            onChange={event => {
+                // handle when clear is clicked and value has been wiped
+                if (event.target.value === "") {
+                    try {
+                        cell.onChange({
+                            ...cell.value,
+                            data: {
+                                ...cell.value.data,
+                                // attempt to reset to cached date
+                                date: date !== undefined ? date : new Date(displayDate),
+                            },
+                        });
+                    } catch (error) {
+                        cell.onChange({
+                            ...cell.value,
+                            data: {
+                                ...cell.value.data,
+                                displayDate: String(error),
+                            },
+                        });
+                    }
+                    return;
+                }
+                cell.onChange({
+                    ...cell.value,
+                    data: {
+                        ...cell.value.data,
+                        // use valueAsNumber because valueAsDate is null for "datetime-local"
+                        // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#technical_summary
+                        date: new Date(event.target.valueAsNumber) ?? cellDataDate,
```

Comment:
Can we get some unit tests around some of this behavior? I know that might be hard but I would appreciate some ideas of what you think should be tested.

Reply:
never too late to add `data-picker-cell.test.tsx` 👍 

The cells package is definitely lacking on the tests front.

Reply:
Cool I will do that!

Reply:
This might be a dumb question but where exactly would the unit test go? I created a date-picker-cell.test.tsx in the cells package. However, when I run a `npm run test`, it's only running the core tests. As a result, what would be the best way organizationally to add this test?

Reply:
Oh uh... heh... oh what a day to be called out as a software developer. Yeah... you would need to add the jest command and setup and ya know what I will add the framework and you can rebase onto the 5.2.2 branch as soon as I add that...